src/rerank/rerank_calculator.cc

```diff
@@ -247,20 +310,18 @@ class RerankCalculator : public CalculatorBase {
             throw std::runtime_error("Batch size mismatch");
 
         std::vector<float> scores;
-        int logits_dim = logits.get_shape()[1];
-
-        if (logits_dim > 1) {
-            // Extract the second column
-            for (int i = 0; i < input_ids.get_shape()[0]; ++i) {
-                float logit = reinterpret_cast<float*>(logits.data())[i * logits_dim + 1];  // TODO: Untested, model has second dimension=1, taken from OpenVINOReranker
-                scores.push_back(1 / (1 + std::exp(-logit)));
-            }
-        } else {
-            // Flatten the logits
-            for (int i = 0; i < input_ids.get_shape()[0]; ++i) {
-                float logit = reinterpret_cast<float*>(logits.data())[i];
-                scores.push_back(1 / (1 + std::exp(-logit)));
-            }
+        scores.resize(actual_batch_size, 0);
+
+        size_t logits_dim = logits.get_shape()[1];
+
+        for (int i = 0; i < input_ids.get_shape()[0]; ++i) {
+            size_t score_index = chunkMapping[i];
+            if (score_index >= actual_batch_size)
+                throw std::runtime_error("score_index out of bounds");  // should never happen
+            float logit = logits_dim > 1 ? reinterpret_cast<float*>(logits.data())[i * logits_dim + 1] : reinterpret_cast<float*>(logits.data())[i];
+            float score = 1 / (1 + std::exp(-logit));
```

Comment:
do we need to calculate it for every chunk or the most likely would be sufficient?

Reply:
what do you mean most likely?
this is exactly what we do - select highest score from all the chunks of given document